### DISCUSSION TITLE
Confusion about the _id of application

### DISCUSSION THREAD

**Discussion by zhengtulymGh** [Budibase/budibase#6672] [Help]
Why all the application use the same _id which value is a constant --- DocumentTypes.APP_METADATA
Why not a self-generated _id, like users' _id.

As you can see, all the created application doc has the same _id that value is  "app_metadata"
```
[
  {
    "_id": "app_metadata",
    "_rev": "106-312f9a67b827064cf3d900102eadc4c2",
    "appId": "app_dev_18b5cdab91094345977e56506c0cd1e4",
    "type": "app",
    "version": "1.0.105-alpha.22",
    "componentLibraries": [
      "@budibase/standard-components"
    ],
    "name": "Employee Timesheet Template_1657626386495",
    "url": "/employee-timesheet-template_1657626386495",
    "instance": {
      "_id": "app_dev_18b5cdab91094345977e56506c0cd1e4"
    },
    "tenantId": "default",
    "status": "development"
  },
  {
    "_id": "app_metadata",
    "_rev": "232-f9c2a4fd228d485da40982fa6be0f87b",
    "appId": "app_dev_306816778be9476eb7db93506e06ffaf",
    "type": "app",
    "version": "1.0.105-alpha.22",
    "componentLibraries": [
      "@budibase/standard-components"
    ],
    "name": "IT Asset Management Software_1657619478820",
    "url": "/it-asset-management-software_1657619478820",
    "instance": {
      "_id": "app_dev_306816778be9476eb7db93506e06ffaf"
    },
    "tenantId": "default",
    "status": "development"
  },
  {
    "_id": "app_metadata",
    "_rev": "82-6618bb3fc73c2a58b74c3ef80570b6e4",
    "appId": "app_dev_34e6c545cfc94fe8aa0133114304db9a",
    "type": "app",
    "version": "1.0.105-alpha.22",
    "componentLibraries": [
      "@budibase/standard-components"
    ],
    "name": "Employee Equipment Checkout Form_1657619594076",
    "url": "/employee-equipment-checkout-form_1657619594076",
    "instance": {
      "_id": "app_dev_34e6c545cfc94fe8aa0133114304db9a"
    },
    "tenantId": "default",
    "status": "development"
  }
]
```

**Comment by aptkingston** [comment]
Hi @zhengtulymGh. This is just the structure that our API uses. You can see the real ID is present in 2 different locations in that payload:
- `appId`
- `instance._id`

You can use either of those fields to reference the unique app IDs.

**Reply by zhengtulymGh** [reply]
Hi @aptkingston. The problem is that due to the use of fixed id, when retrieving data, you sometimes get the wrong data.
For example:
```
const appInfo = await db.get(DocumentTypes.APP_METADATA)
```
 I find that sometimes the wrong app data is returned when visit the published app pages.
As I understand it, CouchDB has a mechanism for replication and conflict resolution when using the same ID.
I'm not sure this mechanism will always return the correct data.

**Reply by aptkingston** [reply]
If you're experiencing a problem, could you open an issue and provide as much information as possible? If you're being served the wrong pages when visiting published apps then that sounds like a serious issue, but not one that I have seen before.

**Reply by zhengtulymGh** [reply]
Hi @aptkingston, There's no problem here, I didn't get the logic straight. Thanks.

**Comment by zhengtulymGh** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，来件已收到，我会尽快回复您。。。。
